backend/common/services/simulation_helper/scene_memory_service.py

```
@@ -0,0 +1,709 @@
+                # Check if memory already exists
+                if scene_key in self.scene_contexts:
+                    return True
+                
+                # Load existing memory from database first
+                try:
+                    await self._load_scene_memory(user_progress_id, scene_id)
+                except Exception as e:
+                    # Check if we're in production environment
+                    if is_production():
+                        # In production, log error and re-raise to ensure proper alerting
+                        logger.error(f"Failed to load scene memory for user_progress_id={user_progress_id}, scene_id={scene_id}: {e}")
```

Ruff (0.14.8)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)